examples/mint-repl/mint-repl.rs

```diff
@@ -185,17 +188,19 @@ fn mk_new_random_mint(threshold: usize, amount: u64) -> Result<MintInfo> {
 /// creates a new mint from an existing SecretKeySet that was seeded by poly.
 fn mk_new_mint(secret_key_set: SecretKeySet, poly: Poly, amount: u64) -> Result<MintInfo> {
     let genesis_pubkey = secret_key_set.public_keys().public_key();
-    let mut mints: Vec<Mint> = Default::default();
+    let mut mints: Vec<Mint<KeyManager>> = Default::default();
 
     // Generate each Mint node, and corresponding NodeSignature. (Index + SignatureShare)
     let mut genesis_set: Vec<(DbcContent, DbcTransaction, (PublicKeySet, NodeSignature))> =
         Default::default();
     for i in 0..secret_key_set.threshold() as u64 + 1 {
-        let key_manager = KeyManager::new(
-            secret_key_set.public_keys().clone(),
-            (i, secret_key_set.secret_key_share(i).clone()),
+        let key_manager = Arc::new(KeyManager::new(
```

Comment:
why does this need to use Arc?

Reply:
Due to line 189 in `mint.rs`:

```rust
    pub fn verifier(&self) -> Verifier<K> {
        Verifier::new(self.key_manager.clone())
    }
```

Reply:
ok, I see there is:

```
pub struct Verifier<K: KeyManager> {
    key_manager: Arc<K>,
}

impl<K: KeyManager> Verifier<K> {
    pub fn new(key_manager: Arc<K>) -> Self {
        Self { key_manager }
    }
```

I just don't understand yet the reason for (a) using Arc in sn_dbc, and (b) exposing/requiring it at API level, ie Verifier::new(km: Arc<K>)

I guess I'm just wondering if we are mixing concerns here.  Maybe this is anticipating changes in the async PR?

Reply:
It's to be able to return a `Verifyer` from the `Mint`., as to pass in something with a limited API to the dbc when confirming valid.
That was the original design, and in the discussion it was said to be better not to pass in `KeyManager` there.

Now that we don't actually hold secret keys, we only hold a reference to something that updates under the hood.
Unless we want to use that thing directly this is sort of the thing I'd say, as we can't have a verifier with a reference to keymanager without it. At least this is a simple way.

But I can just pass in the `KeyManager` straight away, if exposing that fulll api within DBC is less of a problem than Arc.

Reply:
Pushed a commit with that approach now.